### ISSUE TITLE
FMP incredible inaccurate and slow timings 

### ISSUE THREAD

**Issue by lethevimlet** [GoogleChrome/lighthouse#5393]
I was quite happy with the old audit tools, but I see the potential in Lighthouse, anyway here are my 2 main concerns with it.

1) It's a tool that depends on internet connection, this is great to diagnose latency and cache but is horrible two diagnoses code performance which is kind on the first goal of an audit tool.

2) Second and probably related to the first point (unless actually, this is a bug), the performance test is unreliable.

I wrote the simplest web on Earth to prove the concept just a div and no external files.

```
<!DOCTYPE html>
<html>
<head>
    <title>The simplest web on Earth</title>
</head>
<body>
    <div style="height: 3000px; background-color: red; color:white">
        Why Lighthouse slow :(
    </div>
</body>
</html>

```

And here is the shocking results...

![image](https://user-images.githubusercontent.com/2963923/40778786-c43fb070-64d2-11e8-90ec-22a9acb259fd.png)

Apparently painting a div with a red background takes over **2.2 seconds**

That is simply wrong in any possible real scenario. I remember using a Pentium MX on a 56K connection and rendering pages on obsolete browsers in less than 2 seconds... what the hell is going on here Lighthouse are you rendering this on a wooden PC powered by a hamster spinning a wheel XD?

Anyway, hope this helps with the future of lighthouse.

**Comment by patrickhulce** [comment]
> It's a tool that depends on internet connection

If you're auditing a live site, yes using DevTools at all will require an internet connection, but as in your example auditing localhost:8080 does not require an internet connection. What led you to believe this so we can correct it?

> this is great to diagnose latency and cache but is horrible two diagnoses code performance which is kind on the first goal of an audit tool

Perhaps your ideal auditing tool is exclusively focused on code execution performance, but that's not the *first* goal of *all* audit tools :) We provide performance metrics with a mobile-by-default focus, try to identify opportunities for improvement, and check for developer best practices in other non-performance categories.

> Second and probably related to the first point (unless actually, this is a bug), the performance test is unreliable.

There are [lots of reasons](https://docs.google.com/document/d/1BqtL-nG53rxWOI5RO0pItSRPowZVnYJ_gBEQCJ5EeUE/edit) for variability. In Lighthouse v3, there is a new performance engine that mitigates several of these factors and reduces variance by ~50%, but it is not and never will be perfect. If completely eliminating variance is important to your use case, I'd encourage you to familiarize yourself with the sources of variance and try to create a testing environment that addresses all of them.

> That is simply wrong in any possible real scenario.

Not exactly since this is precisely how long it took to load on your machine with the default throttling settings ;) but yes it's odd for the first paint to have been delayed by 2s for such a simple page. This happens every time for you? Do you have any extensions/anti-virus that might be interfering with page load?

**Comment by lethevimlet** [comment]
First thanks for the kind and quick response. Addressing the following:

> What led you to believe this so we can correct it?

I disabled the LAN interface and restarted chrome, then run Lighthouse performance test for a localhost server and Lighthouse refused to run, it got stock each time I did this, running on a Win10, Chrome 66 environment, so maybe it seems to be actually a bug.

>  I'd encourage you to familiarize yourself with the sources of variance and try to create a testing environment that addresses all of them.

Using performance tab on devTools yields a stable result, so I imagined Lighthouse would be more or less the same especially when working in such a minimalistic environment.

> but yes it's odd for the first paint to have been delayed by 2s for such a simple page. This happens every time for you? Do you have any extensions/anti-virus that might be interfering with page load?

Yes, this happens always, even running a single tab and without extensions. The machine is quite powerful so even with a high "x" throttling should yield better results. Once again as a comparison with performance dev tools tab, I can estimate the FMP in less than 150ms (no throttling).

I provided the code so anyone can test it and compare results.

**Comment by patrickhulce** [comment]
> I disabled the LAN interface and restarted chrome, then runned Lighthouse performance test for a localhost server and Lighthouse refused to run, it got stuck each time I did this

Was it the first time you ever used Lighthouse? It is a remote module (not directly bundled with Chrome) so the first time it's used it will need to be downloaded but after that it is available offline. This is the same as a few other DevTools features (like device art/emulation, etc). If it wasn't the first time and every time you disconnect this happens, this is definitely a bug we haven't seen so a separate bug report would be much appreciated :)

> Using performance tab on dev tools yields a stable result, so I imagined Lighthouse would be more or less the same especially when working in such a minimalistic environment.

Lighthouse is backed by the same exact tooling (in fact if you open up Chrome Canary and run there you'll see the "View Trace" button that will take you over to the Performance tab with the underlying information). If you're not seeing any variance there, you shouldn't be seeing the variance in Lighthouse. When I run that page I see first paints of about ~600ms which is basically the minimum possible value with 3G throttling. If you save the trace of 2.2s happening in Canary we can take a better look.

audit of https://output.jsbin.com/mubisobeqo
![image](https://user-images.githubusercontent.com/2301202/40813591-854a6f9a-64f0-11e8-826f-8f1bae528fc7.png)

**Comment by AymenLoukil** [comment]
@lethevimlet Try to run Lighthouse in Incognito mode.

**Comment by paulirish** [comment]
resolved in Lighthouse 3.0+
see https://github.com/GoogleChrome/lighthouse/issues/5393#issuecomment-393715011

**Comment by lethevimlet** [comment]
I can confirm the issue is resolved 👍
